packages/browser-agent-core/features/spa/instrument/index.js

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2020 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+*/
+import { wrapMutation, wrapPromise, wrapHistory, wrapEvents, wrapTimer, wrapFetch, wrapXhr, wrapJson } from '../../../common/wrap'
+import { eventListenerOpts } from '../../../common/event-listener/event-listener-opts'
+import { FeatureBase } from '../../../common/util/feature-base'
+import { getRuntime } from '../../../common/config/config'
+import { now } from '../../../common/timing/now'
+
+var START = '-start'
+var END = '-end'
+var BODY = '-body'
+var FN_START = 'fn' + START
+var FN_END = 'fn' + END
+var CB_START = 'cb' + START
+var CB_END = 'cb' + END
+var JS_TIME = 'jsTime'
+var FETCH = 'fetch'
+var ADD_EVENT_LISTENER = 'addEventListener'
+
+var win = window
+var location = win.location
+
+export class Instrument extends FeatureBase {
+    constructor(agentIdentifier) {
+        super(agentIdentifier)
+        const agentRuntime = getRuntime(this.agentIdentifier);
+        // loader.xhrWrappable will be false in chrome for ios, but addEventListener is still available.
+        // sauce does not have a browser to test this case against, so be careful when modifying this check
+        if (!win[ADD_EVENT_LISTENER] || !agentRuntime.xhrWrappable || agentRuntime.disabled) return
+        agentRuntime.features.spa = true;
+
+        // console.log("initialize spa instrument!", agentIdentifier)
+
+        let depth = 0
+        let startHash
+
+        const tracerEE = this.ee.get('tracer')
+        const jsonpEE = wrapJson(this.ee)
+        const promiseEE = wrapPromise(this.ee)
+        const eventsEE = wrapEvents(this.ee)
+        const timerEE = wrapTimer(this.ee)
+        const xhrEE = wrapXhr(this.ee)
+        const fetchEE = wrapFetch(this.ee)
+        const historyEE = wrapHistory(this.ee)
+        const mutationEE = wrapMutation(this.ee)
+
```

Comment:
Importing and running `xhr/instrument` was removed, which will cause problems if the Ajax feature is not required to be imported or named as an explicit dependency of SPA:

https://github.com/newrelic/newrelic-browser-agent/blob/e0addb7c19e52429cc17b50a4e6f01892b3c1fc1/feature/spa/instrument/index.js#L38

We should re-add this or create a dependency so we're at least warned if Ajax is ever disabled